File: Practice/Task4 python/02/task2.py
#!/usr/bin/python3
import pathlib
import sys
import os
import shutil

if (sys.argv[1] == "--help"):
	print("Usage: task2.py DIR DIR_TO .EXT")
	print("Copy all files that have EXT extention from DIR to DIR_TO")

elif (sys.argv[1] == "--test"):
	os.system("./test2.bash")

elif (len(sys.argv) == 4):
        dir_from = sys.argv[1]
        dir_to = sys.argv[2]
        ext = sys.argv[3]
        for path, subdirs, files in os.walk(dir_from):
                for file in files:
                        if pathlib.Path(file).suffix == ext:
                                shutil.copy(os.path.join(path, file), os.path.join(dir_to, file))

File: Practice/Task8 .net Core Linux self-contained/02/Task2.cs
using System;
using System.IO;
using System.Diagnostics;

namespace Task2
{
    internal class Program
    {	    	
        static void Main(string[] args)
        {
		if (!(args.Length == 3 || args.Length == 1))
		{
			Console.WriteLine("Task2: invalid arguments quantity");
			Console.WriteLine("Try 'Task2 --help' for more information");
		}
		else if(Equals(args[0], "--help"))
		{
			Console.WriteLine("Usage: Task2 DIR DIR_TO .EXT");
      			Console.WriteLine("Copy all files that have EXT extention from DIR to DIR_TO");
		}
		else if(Equals(args[0], "--test"))
		{
			var psi = new ProcessStartInfo();
			psi.FileName = "./test2.bash";
			psi.UseShellExecute = false;
			psi.CreateNoWindow = true;
			using var process = Process.Start(psi);
			if (process != null) process.WaitForExit();
		}	
		else
		{		 	
		 	string sourcePath = args[0];
			string targetPath =  args[1]; 
			var extensions = new[] { args[2] }; 
			var files = (from file in Directory.EnumerateFiles(sourcePath, "*", SearchOption.AllDirectories)
			where extensions.Contains(Path.GetExtension(file), StringComparer.InvariantCultureIgnoreCase) 
			select new 
			{
				Source = file, 
				Destination = Path.Combine(targetPath, Path.GetFileName(file))
			});
			foreach(var file in files)
			{
			  File.Copy(file.Source, file.Destination);
			}		
		}
        }
    }
}
